Documentation/dash-hld.md

```diff
@@ -0,0 +1,187 @@
+
+
+### DASH container
+ 
+The SDN controller communicates with a DASH device through a **[gNMI](https://github.com/Azure/DASH/wiki/Glossary#gnmi) endpoint** served by a new DASH SDN agent **running inside a new SONiC DASH container**.  
+
+In summary:
+- The DASH container translates SDN configuration modeled in gNMI into **SONiC DB** objects.
+- The **SONiC orchagent** inside the Switch State Service (SWSS) Container will be enhanced to translate these objects into **SAI_DB objects**, including the new **DASH-specific SAI objects**. 
+- An **enhanced syncd** will then configure the dataplane using the **vendor-specific SAI library**.
+
+A **gNMI schema** will manage the following DASH services: 
+- Elastic Network Interface (ENI)
+- Access Control Lists (ACLs) 
+- Routing and mappings
+- Encapsulations 
+- Other  
+
+See [TODO] for DASH gNMI schema.
+
+#### Multiple DPUs device
+
+In the case of a multiple DPUs device the following applies:
+
+- Each DPU provides a gNMI endpoint for SDN controller through a unique IP address. 
+- An appliance or smart switch containing multiple DPUs therefore contains multiple gNMI endpoints for SDN controller, and the controller treats each DPU as a separate entity. 
+- To conserve IPv4 addresses, such an appliance or switch _might_ contain a proxy (NAT) function to map a single IP address:port combination into multiple DPU endpoints with unique IPv4 addresses.  
+- No complex logic will run on the switches (switches do not have a top-level view of other/neighboring switches in the infrastructure). 
+
+
+### Switch State Service (SWSS)
+The SWSS container comprises many daemons which operate on conceptual SONIC config objects across several databases. DASH affects the following daemons, as follows:
+* `orchagent`, which translates `xx_DB` objects into `ASIC_DB` objects, must be enhanced to manage new DASH overlay objects, such as ACL1,2,3 rules, ENI mappings, etc. The `orchagent` has to manage both representations of SONiC objects (xx_DB and ASIC_DB) and translates between them bidirectionally as required.
+* `syncd`, which translates `ASIC_DB` conceptual objects into vendor SAI library API calls, must likewise be enhanced to handle new DASH SAI objects.
+
+### Switch Abstraction Interface (SAI) DASH
+
+The Switch Abstraction Interface (SAI) is a common API that is supported by many switch ASIC vendors. SONiC uses SAI to program the ASIC. This enables SONiC to work across multiple ASIC platforms naturally. DASH uses a combination of traditional SAI headers and new DASH pipeline-specific headers. Vendors must implement this interface for their DASH devices. This is the primary integration point of DASH devices and the SONiC stack. It will be rigorously tested for performance and conformance. See [DASH Testing documentation](https://github.com/Azure/DASH/tree/main/test).
```

Comment:
```suggestion
The Switch Abstraction Interface (SAI) is a common API that is supported by many switch ASIC technology suppliers. SONiC uses SAI to program the ASIC. This enables SONiC to work across multiple ASIC platforms naturally. DASH uses a combination of traditional SAI headers and new DASH pipeline-specific headers. Technology suppliers must implement this interface for their DASH devices. This is the primary integration point of DASH devices and the SONiC stack. It will be rigorously tested for performance and conformance. See [DASH Testing documentation](https://github.com/Azure/DASH/tree/main/test).
```